src/Form/Input.js

```diff
@@ -52,17 +52,27 @@ const StyledInput = createComponent({
 
 const StyledTextArea = StyledInput.withComponent('textarea');
 
+const validateValueProp = (props, propName, componentName) => {
+  if (props.type === 'number' && typeof props.value !== 'number') {
```

Comment:
lets generalize this to `propName` like below: `... && typeof props[propName] !== 'number'`